src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -274,6 +275,40 @@ internal BinaryClassificationTrainers(BinaryClassificationCatalog catalog)
                 Evaluate(x.Scores, labelColumn), x.Scores, x.Fold)).ToArray();
         }
 
+        /// <summary>
+        /// Change threshold for binary model.
+        /// </summary>
+        /// <typeparam name="TModel">An implementation of the <see cref="IPredictorProducing{TResult}"/></typeparam>
+        /// <param name="chain">Chain of transformers.</param>
+        /// <param name="threshold">New threshold.</param>
+        /// <returns></returns>
+        public TransformerChain<BinaryPredictionTransformer<TModel>> ChangeModelThreshold<TModel>(TransformerChain<BinaryPredictionTransformer<TModel>> chain, float threshold)
+            where TModel : class
+        {
+            if (chain.LastTransformer.Threshold == threshold)
+                return chain;
+            List<ITransformer> transformers = new List<ITransformer>();
+            var predictionTransformer = chain.LastTransformer;
```

Comment:
>chain.LastTransformer [](start = 40, length = 21)

I don't like the assumption that the predictor is the last one, it might not be. Minimally, we should check, but that is not safe either. 

IMO the only API existing for this should be the second one.

Reply:
That's a good point @sfilipi. I think you're probably right about this.

---
In reply to: [266034490](https://github.com/dotnet/machinelearning/pull/2969#discussion_r266034490) [](ancestors = 266034490)

Reply:
My idea was to provide helper function for user since work with transform chain is kinda painful, at least from my point.
But I can have one method.

---
In reply to: [268206172](https://github.com/dotnet/machinelearning/pull/2969#discussion_r268206172) [](ancestors = 268206172,266034490)